### PR TITLE
Update Project template

### LINKED ISSUE #2494: Include `serde` dependency in template for `linera project new`
# Motivation

Most (if not all) applications will require implementing at least a custom operation, response, message and/or query type. These types need to implement `Serialize` and `Deserialize`. However, `serde` is not included by default in projects created with `linera project new`. There's no gain from this, because `linera-sdk` already depends on `serde`. Not having it gives new users an extra "usability papercut" once they realize that they should add `serde` and derive the appropriate traits.

# Proposal

Include `serde` as a dependency by default in newly created projects. As an example, change the `ContractAbi::Operation` type to be an empty `enum` (or with a single variant) that derives `Serialize` and `Deserialize`.

# Proposal

The template is located in [`linera-service/template`](https://github.com/linera-io/linera-protocol/tree/761bb2ee4d8189e639c533cecdc10a8a2c5f6afe/linera-service/template).

### PR DESCRIPTION
**Pull request by Nirajsah**
## Closes #2494 #2492 #2488

**Summary:** This pull request updates the necessary files to generate required template files to get started, when executing the command ```linera project new project-name```.

**Changes Made:**

- Added required dependencies to facilitate project initialization.
- Included unit test files to provide examples of test implementation. In ```contract.rs``` and ```service.rs```.
- Added an additional `Integration`  test file to evaluate ```Query``` and ```Operation```, ensuring proper functionality.

**Testing:** All test cases have been successfully executed.

### CODE REVIEW

**Comment by Nirajsah** [comment]
i don't understand why the test failed. i've tested it locally
some tests are failing in ```linera-witty/tests/common```. the same tests failed in two diff branches one with the commit, and one without the commit. i don't understand why?

**Comment by afck** [comment]
@Nirajsah: Thank you!
It looks like `test_project_publish` in `linera-service/tests/linera_net_tests.rs` is failing because it still uses `&()` as an instantiation argument. You changed it to a number, so that test needs to be updated.

**Comment by Nirajsah** [comment]
> @Nirajsah: Thank you! It looks like `test_project_publish` in `linera-service/tests/linera_net_tests.rs` is failing because it still uses `&()` as an instantiation argument. You changed it to a number, so that test needs to be updated.

:heart:  thanks for reviewing. running test locally inside `linera-service` doesn't show any error or test failure. i ran ```cargo test``` in linera-service. is there some other tests to identify such issues. 

```
let application_id = client
        .publish_and_create::<EthereumTrackerAbi, (), InstantiationArgument>(
            contract,
            service,
            u64, #changed here. &()
            &argument,
            &[],
            None,
        )
        .await?;
```
also just updating the code to have ```u64``` will solve the issue, i assume. cuz i don't see any errors while running the test, so its hard to tell.

**Comment by afck** [comment]
Yes, `cargo test` doesn't run the tests that run against a network by default. To run them, start the storage server and enable the `storage-service` flag:

```
export LINERA_STORAGE_SERVICE=127.0.0.1:1235
cargo run --bin linera-storage-server -- memory --endpoint $LINERA_STORAGE_SERVICE &
cargo test --features storage-service -p linera-service
```

**Comment by Nirajsah** [comment]
> @Nirajsah: Thank you! It looks like `test_project_publish` in `linera-service/tests/linera_net_tests.rs` is failing because it still uses `&()` as an instantiation argument. You changed it to a number, so that test needs to be updated.

i tried searching for `test_project_publish` in `linera-service/tests/linera_net_tests.rs` found it was in `linera-service/tests/local_net_tests.rs`. i've updated the instantiation argument to `&0`

**Review by afck — COMMENTED** [review]
I'm not sure we want the whole counter application to be generated with `project new`. It's meant to be a template, not an example. So I'd replace the implementation and test code with `todo!()`s, and give the test a more generic name. (Just `test`?)

In `test_project_publish`, I'd use the actual `examples/counter` directory rather than the template.

(Happy to wait for others to chime in, though. That's just my opinion.)

**Comment by Nirajsah** [comment]
@afck 
thanks for the suggestions, i'll update everything.

**Review by afck — APPROVED** [review]

File: linera-service/template/contract.rs.template [hunk 1]
```diff
@@ -33,13 +35,68 @@ impl Contract for {project_name}Contract {{
...
+    contract
+        .instantiate(initial_value)
+        .now_or_never()
+        .expect("Initialization of application state should not await anything");
+
+    assert_eq!(*contract.state.value.get(), initial_value);
+
+    contract
+  }} 
+}}
```
[the remark this answers is not in the document]
Comment (ma2bd) [reply linera-service/template/contract.rs.template:101]: We could test in CI that the generated files are formatted.

[the remark this answers is not in the document]
Comment (afck) [reply linera-service/template/contract.rs.template:101]: I created https://github.com/linera-io/linera-protocol/issues/2730 for this. I don't think it's a blocker, so I'll merge this one.

**Pull request merged**